### PR TITLE
use a is_windows variable instead of function

### PR DESCRIPTION
**Pull request by glepnir**
define a variable `is_windows` replace regex function.

### CODE REVIEW

File: runtime/lua/vim/uri.lua [hunk 1]
```diff
@@ -3,7 +3,7 @@
 -- https://tools.ietf.org/html/rfc2732
 -- https://tools.ietf.org/html/rfc2396
 
-
+local is_windows = jit.os == 'Windows'
```
Comment (vigoux) [inline runtime/lua/vim/uri.lua:6]: When not using `LuaJIT`, `jit` will be null, causing error.
This is a good idea though, but we should provide a fallback when not using `jit`.

Comment (glepnir) [reply runtime/lua/vim/uri.lua:6]: Is it possible that jit will be removed in the future?

Comment (glepnir) [reply runtime/lua/vim/uri.lua:6]: What about use this `is_windows = vim.loop.os_uname().version:match("Windows")`

Comment (glepnir) [reply runtime/lua/vim/uri.lua]: I had update.Please check. Thanks.

File: runtime/lua/vim/uri.lua [hunk 2]
```diff
@@ -84,7 +84,7 @@ local function uri_to_fname(uri)
   end
   uri = uri_decode(uri)
   -- TODO improve this.
-  if is_windows_file_uri(uri) then
+  if is_windows then
```
Comment (vigoux) [inline runtime/lua/vim/uri.lua:87]: Maybe this `is_windows_file_uri` can be removed now ?

File: runtime/lua/vim/uri.lua (hunk 1 above)
Comment (vigoux) [reply runtime/lua/vim/uri.lua]: Yeah I quite like this version.

Comment (vigoux) [reply runtime/lua/vim/uri.lua]: Tests are failing though

File: runtime/lua/vim/uri.lua (hunk 2 above)
Comment (glepnir) [reply runtime/lua/vim/uri.lua:82]: Yes.removed.

File: runtime/lua/vim/uri.lua [hunk 3]
```diff
@@ -3,7 +3,7 @@
 -- https://tools.ietf.org/html/rfc2732
 -- https://tools.ietf.org/html/rfc2396
 
-
+local is_windows = vim.loop.os_uname().version:match("Windows")
```
Comment (vigoux) [inline runtime/lua/vim/uri.lua:6]: Maybe you meant
```suggestion
local is_windows = vim.loop.os_uname().sysname:match("Windows")
```

Comment (glepnir) [reply runtime/lua/vim/uri.lua:6]: Are there any differences between them?

Comment (vigoux) [reply runtime/lua/vim/uri.lua:6]: On my end `sysname` is `Linux` to I expect it to be `Windows` on windows machines.

Comment (glepnir) [reply runtime/lua/vim/uri.lua]: Sure. changed.

**Comment by vigoux** [comment]
The `uri` tests are all failing, which is weird, I really suggest to run `TEST_FILE=test/functional/lua/uri_spec.lua make test` to check the output and fix those tests, that should allow you to quickly iterate.

**Comment by h-michael** [comment]
@glepnir 
You should fix luacheck results, not test.

https://travis-ci.org/github/neovim/neovim/jobs/725212536#L2335-L2339

**Comment by glepnir** [comment]
check this test file `test/functional/lua/uri_spec.lua`. I found the test value Uri are all hard-coded incoming functions for testing. If judge whether it is windows through variable, it will read the current operating system, which will cause the test to fail

**Pull request closed without merging**